[ao][docs] Add description of qconfig and qengine to quantization page

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63270
* __->__ #63582

Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

raghuramank10000 committed:
[ao][docs] Add description of qconfig and qengine to quantization page

Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on docs/source/quantization.rst:
```diff
@@ -35,6 +35,13 @@ that perform all or part of the computation in lower precision. Higher-level
+1. Quantization Config (Qconfig): Specifies how weights and activations are to be quantized. Qconfig is needed to create a quantized model.
+2. Backend: Refers to kernels that support quantization, usually with different numerics.
+3. Quantization engine (torch.backends.quantization.engine): When a quantized model is executed, the qengine specifies which
+backed is to be used for execution. It is important to ensure that the qengine is consistent with the Qconfig.
```
typo: backed -> backend

supriyar replied:
Is the new line here required?

supriyar commented on docs/source/quantization.rst:
```diff
@@ -45,7 +52,8 @@ Today, PyTorch supports the following backends for running quantized operators e
 
-The corresponding implementation is chosen automatically based on the PyTorch build mode.
+The corresponding implementation is chosen automatically based on the PyTorch build mode, though users
+have the option to override this by setting `torch.backends.quantization.engine` to `fbgemm` or `qnnpack`.
```
maybe adding a small code-block here specifying how to override would be helpful

supriyar reviewed (approved):
looks good, left a few comments inline

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented:
@raghuramank100 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30658656).

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "[ao][docs] Add description of qconfig and qengine to quantization page"


Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

[ghstack-poisoned]
